dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/DimensionalItemObject.java

```diff
@@ -91,4 +91,18 @@
      * should be aggregated across multiple values.
      */
     TotalAggregationType getTotalAggregationType();
+
+    /**
+     * Gets a Period Offset: the offset can be applied within an Indicator formula
+     * in order to "shift" the query period by the offset value (e.g. Jan 2020 with
+     * offset 1 becomes Feb 2020). An offset with value 0 means no offset.
+     * 
+     * @return an int.
+     */
+    int getPeriodOffset();
+
+    /**
+     * Set the Period Offset to 0
+     */
+    void resetPeriodOffset();
```

Comment:
This gives me a little sense of mutability issues. Perhaps one could instead close the DimensionalItemObject somehow and keep it immutable, or move mutable properties another place. I know that has drawbacks too as discussed.

Reply:
Fix. Yes, beside this method no longer needed, since the `periodOffset` field is now `transient` so the field is not serialized.